### PR TITLE
Feat/upgrade to rest

### PR DESCRIPTION
**Pull request by ffiebig**
Replaced SOAP services with REST version

### CODE REVIEW

File: Controller/Transaction/CommitWebpayM22.php [hunk 1]
```diff
@@ -150,44 +152,64 @@ protected function toRedirect($url, $data)
...
             <b>Fecha de la Transacci&oacute;n: </b>" . date('d-m-Y', strtotime($transactionResult['transactionDate'])) . "<br>
             <b>Hora de la Transacci&oacute;n: </b>" . date('H:i:s', strtotime($transactionResult['transactionDate'])) . "<br>
-            <b>Tarjeta: </b>**** **** **** {$transactionResult['cardDetail']['cardNumber']}<br>
-            <b>C&oacute;digo de autorizacion: </b>{$transactionResult['detailOutput']['authorizationCode']}<br>
+            <b>Tarjeta: </b>**** **** **** {$transactionResult['cardDetail']['card_number']}<br>
+            <b>C&oacute;digo de autorizacion: </b>{$transactionResult['authorizationCode']}<br>
             <b>Tipo de Pago: </b>{$paymentType}<br>
             <b>Tipo de Cuotas: </b>{$tipoCuotas}<br>
-            <b>N&uacute;mero de cuotas: </b>{$transactionResult['detailOutput']['sharesNumber']}
+            <b>N&uacute;mero de cuotas: </b>{$transactionResult['installmentsNumber']}
```
Comment (gdespirito) [inline Controller/Transaction/CommitWebpayM22.php]: Maybe you can put the installment amount next to the installments number :)

File: Model/Config/Environment.php [hunk 2]
```diff
@@ -9,8 +9,8 @@ class Environment implements \Magento\Framework\Option\ArrayInterface {
      * @return array
      */
     public function toOptionArray() {
-        return [['value' => 'INTEGRACION', 'label' => __('INTEGRACION')],
-                ['value' => 'PRODUCCION', 'label' => __('PRODUCCION')]];
+        return [['value' => 'TEST', 'label' => __('Integración (Pruebas)')],
```
Comment (gdespirito) [inline Model/Config/Environment.php:12]: Nice

File: etc/adminhtml/system.xml [hunk 3]
```diff
@@ -1,66 +1,71 @@
...
+                    <!-- Commerce Code -->
+                    <field id="commerce_code" translate="label" type="text" sortOrder="50" showInDefault="1" showInWebsite="1" showInStore="0">
+                        <label>Código de Comercio</label>
+                        <comment>Identificador único del comercio, asegurate de que comience con 5970</comment>
+                    </field>
 
+                    <!-- Api Key -->
+                    <field id="api_key" translate="label" type="textarea" sortOrder="60" showInDefault="1" showInWebsite="1" showInStore="0">
+                        <label>API Key</label>
+                        <comment>Llave secreta utilizada para conectarse a la API, asegurate de no compartir su contenido!</comment>
```
Comment (gdespirito) [inline etc/adminhtml/system.xml]:
```suggestion
                        <comment>Llave secreta utilizada para conectarse a la API ¡asegúrate de no compartir su contenido!</comment>
```

**Comment by gdespirito** [comment]
I do not see any changes to the "verificar conexión" test on diagnostic modal. Is that working?

**Comment by ffiebig** [comment]
>I do not see any changes to the "verificar conexión" test on diagnostic modal. Is that working?

Yeah, it is, what changed is underneath

**Review by gdespirito — APPROVED** [review]
Great work 🎉

**Pull request merged**